src/Symfony/Component/DependencyInjection/Compiler/ResolveInstanceofConditionalsPass.php

```diff
@@ -149,6 +149,11 @@ private function processDefinition(ContainerBuilder $container, string $id, Defi
                 ->setAbstract(true);
         }
 
+        if ($definition->isSynthetic()) {
+            // Ignore container.excluded tag on synthetic services
```

Comment:
Not sure the comment is needed 😃

Reply:
I added it to make it more explicit that this is the desired behavior.